### PR TITLE
feat(menu): init

### PR DESCRIPTION
**Pull request by OysterD3**

### CODE REVIEW

File: components/menu/src/Menu.svelte [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  import { slide } from "svelte/transition";
+  import { getNodeAttribute } from "@wetix/utils";
+  import type { MenuItems } from "../types";
+
+  let className = "";
+  export { className as class };
+
+  export let items: MenuItems[] = [];
+  export let level = [];
+  export let onSelectMenu = (_) => {};
```
Comment (si3nloong) [inline components/menu/src/Menu.svelte:11]: i think should replace this with `createEventDispatcher`, use `on:change` event

File: components/menu/src/Menu.svelte [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+  const handleSelectMenu = (e: Event) => {
+    const data = JSON.parse(getNodeAttribute(e, "data-item"));
+    onSelectMenu(data);
+  };
+
+  const isActive = (href: string): boolean =>
+    `${window.location.pathname}${window.location.hash}${window.location.search}` === href;
```
Comment (si3nloong) [inline components/menu/src/Menu.svelte:19]: we shouldn't handle the location path, should leave it to user to handle using `on:change` event

File: components/menu/src/Menu.svelte [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+      .responsive-ui-menu__control {
+        transition: all 0.5s;
+        width: 20px;
+        height: 20px;
+        color: #999999;
+        margin-left: auto;
+      }
+
+      &:not(.collapsed) {
```
Comment (si3nloong) [inline components/menu/src/Menu.svelte:53]: attribute name should `--collapsed` right?

File: components/menu/src/Menu.svelte [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        .responsive-ui-menu__control {
+          transform: rotate(90deg);
+        }
+      }
+
+      &.disabled {
```
Comment (si3nloong) [inline components/menu/src/Menu.svelte:59]: attribute name should `--disabled` right?

File: components/menu/src/Menu.svelte [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        cursor: not-allowed !important;
+        opacity: 0.8;
+      }
+
+      &.active {
```
Comment (si3nloong) [inline components/menu/src/Menu.svelte:64]: attribute name should `--active` right?

File: components/menu/types/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import type { SvelteComponentTyped } from "svelte/internal";
+
+export interface MenuItems {
+  title: string;
+  value: any;
+  disabled?: boolean;
+  href?: string;
```
Comment (si3nloong) [inline components/menu/types/index.d.ts:7]: remove this

File: components/menu/package.json [hunk 7]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "@responsive-ui/menu",
   "version": "0.9.1",
-  "description": "> TODO: description",
-  "author": "Si3nLoong <sianloong90@gmail.com>",
-  "homepage": "https://github.com/wetix/responsive-ui#readme",
+  "description": "A menu component for responsive-ui",
+  "author": "Oyster Lee <oysterd3@gmail.com>",
```
Comment (si3nloong) [inline components/menu/package.json:5]: please put your github link as well

**Review by si3nloong — CHANGES_REQUESTED** [review]
Overall, is good, but details part is not. (Sometime it's not about how fast you can complete your task, is about how well you can complete your task)

For example, i didn't see any localStorage is using for restoring the menu state.

File: components/menu/src/Menu.svelte (hunk 2 above)
Comment (OysterD3) [reply components/menu/src/Menu.svelte]: Replaced with `isActive` attribute in items.

**Pull request merged**